src/devtools.js

```diff
@@ -0,0 +1,434 @@
+/* global __REACT_DEVTOOLS_GLOBAL_HOOK__ */
+
+import options from './options';
+import { ATTR_KEY } from './constants';
+import { isFunctionalComponent } from './vdom/functional-component';
+import { Component } from './component';
+
+/** Property on functions used for the component name. */
+let componentNameProperty = 'name';
+
+/**
+ * Override the default property on functional component wrappers which
+ * the function name is saved to.
+ */
+export function _setComponentNameProperty(name) {
+	componentNameProperty = name;
+}
+
+/**
+ * Return a ReactElement-compatible object for the current state of a preact
+ * component.
+ */
+function createReactElement(component) {
+	return {
+		type: component.constructor,
+		key: component.key,
+		ref: null, // Unsupported
+		props: component.props
+	};
+}
+
+/**
+ * Create a ReactDOMComponent-compatible object for a given DOM node rendered
+ * by preact.
+ *
+ * This implements the subset of the ReactDOMComponent interface that
+ * React DevTools requires in order to display DOM nodes in the inspector with
+ * the correct type and properties.
+ *
+ * @param {Node} node
+ */
+function createReactDOMComponent(node) {
+	const childNodes = node.nodeType === Node.ELEMENT_NODE ?
+		Array.from(node.childNodes) : [];
+
+	const isText = node.nodeType === Node.TEXT_NODE;
+
+	return {
+		// --- ReactDOMComponent interface
+		_currentElement: isText ? node.textContent : {
+			type: node.nodeName.toLowerCase(),
```

Comment:
Ah, I didn't know about [.localName](https://developer.mozilla.org/en-US/docs/Web/API/Element/localName). That's only been introduced in Firefox very recently, so I'd be inclined to stick with `toLowerCase()` for the moment and avoid the hassle of polyfilling this. Preserving the case is not that important, it just looks nicer in the devtools if the tag names don't get upper-cased.